Userland/Applications/TextEditor/main.cpp

```diff
@@ -88,13 +88,51 @@ int main(int argc, char** argv)
     text_widget.initialize_menubar(menubar);
     app->set_menubar(menubar);
 
-    if (file_to_edit)
-        text_widget.open_file(file_to_edit);
-    else
+    // If a colon exists in the file name, the user may have specified a line and column number.
+    // First check if a file exists with a name including the colon(s). If so, open it.
+    // Otherwise, split apart the colon(s) from the file name, open the file, and jump to the specified line/column
+
+    if (file_to_edit) {
+        if (StringView(file_to_edit).contains(':')) {
+            if (Core::File::exists(file_to_edit)) {
+                // A file exists with a name including the colon(s), open it.
+                text_widget.open_file(file_to_edit);
+            } else {
+                // A file doesn't exist with a name including the colon(s), maybe the user entered line/column coordinates?
+                auto file_name_vector = StringView(file_to_edit).split_view(':');
+                auto file_name_vector_size = file_name_vector.size();
+                if (!file_name_vector.is_empty()) {
+                    auto file_name = file_name_vector.at(0);
+                    if (file_name_vector_size >= 3) {
```

Comment:
What about the other components?
`foo:bar.cpp:10:42` would open `foo` here.

Maybe invent a `split_end_limit()` and use that? Or just join the parts back with a colon

Reply:
Sorry for taking so long to get back to this! I rewrote it to use regex instead, which solves this issue. Now you're able to have colons as part of the file name without any issues.
I think regex is a better solution anyways compared to the somewhat hacky if-else statements that I had before.

Reply:
I personally find the regex solution more clear (and indeed, it does handle this edge case correctly), so that's a lgtm from me.